File: calculator.cpp
#include <iostream>
#include <cmath>
#include <stdexcept>
using namespace std;

void showMenu() {
    cout << "Calculator\n";
    cout << "1. Addition\n2. Subtraction\n3. Multiplication\n";
    cout << "4. Division\n5. Power\n6. Square Root\n";
    cout << "7. Logarithm\n8. Sine\n9. Cosine\n10. Tangent\n";
    cout << "0. Exit\n";
    cout << "Choose an option: ";
}

void performCalculation(int choice) {
    double a, b;
    switch (choice) {
        case 1:
            cout << "Enter two numbers: ";
            cin >> a >> b;
            cout << "Result: " << a + b << endl;
            break;
        case 2:
            cout << "Enter two numbers: ";
            cin >> a >> b;
            cout << "Result: " << a - b << endl;
            break;
        case 3:
            cout << "Enter two numbers: ";
            cin >> a >> b;
            cout << "Result: " << a * b << endl;
            break;
        case 4:
            cout << "Enter numerator and denominator: ";
            cin >> a >> b;
            if (b == 0) throw runtime_error("Division by zero error!");
            cout << "Result: " << a / b << endl;
            break;
        case 5:
            cout << "Enter base and exponent: ";
            cin >> a >> b;
            cout << "Result: " << pow(a, b) << endl;
            break;
        case 6:
            cout << "Enter a number: ";
            cin >> a;
            if (a < 0) throw runtime_error("Square root of negative number!");
            cout << "Result: " << sqrt(a) << endl;
            break;
        case 7:
            cout << "Enter a number: ";
            cin >> a;
            if (a <= 0) throw runtime_error("Logarithm of non-positive number!");
            cout << "Result: " << log(a) << endl;
            break;
        case 8:
            cout << "Enter an angle (in radians): ";
            cin >> a;
            cout << "Result: " << sin(a) << endl;
            break;
        case 9:
            cout << "Enter an angle (in radians): ";
            cin >> a;
            cout << "Result: " << cos(a) << endl;
            break;
        case 10:
            cout << "Enter an angle (in radians): ";
            cin >> a;
            cout << "Result: " << tan(a) << endl;
            break;
        default:
            cout << "Invalid choice!" << endl;
    }
}

int main() {
    int choice;
    do {
        showMenu();
        cin >> choice;
        if (choice == 0) break;
        try {
            performCalculation(choice);
        } catch (exception &e) {
            cout << "Error: " << e.what() << endl;
        }
        cout << endl;
    } while (choice != 0);
    cout << "Goodbye!" << endl;
    return 0;
}

File: calculator.rs
use std::io;

fn show_menu() {
    println!("Calculator");
    println!("1. Addition");
    println!("2. Subtraction");
    println!("3. Multiplication");
    println!("4. Division");
    println!("5. Power");
    println!("6. Square Root");
    println!("7. Logarithm");
    println!("8. Sine");
    println!("9. Cosine");
    println!("10. Tangent");
    println!("0. Exit");
}

fn main() {
    loop {
        show_menu();
        println!("Choose an option: ");
        let mut choice = String::new();
        io::stdin().read_line(&mut choice).unwrap();
        let choice: i32 = match choice.trim().parse() {
            Ok(num) => num,
            Err(_) => {
                println!("Invalid input. Please enter a number.");
                continue;
            }
        };

        if choice == 0 {
            println!("Goodbye!");
            break;
        }

        match choice {
            1..=10 => {
                let mut input1 = String::new();
                let mut input2 = String::new();
                println!("Enter the first number: ");
                io::stdin().read_line(&input1).unwrap();
                let num1: f64 = input1.trim().parse().unwrap_or_else(|_| {
                    println!("Invalid number!");
                    0.0
                });

                if choice <= 5 {
                    println!("Enter the second number: ");
                    io::stdin().read_line(&input2).unwrap();
                }

                let num2: f64 = if choice <= 5 {
                    input2.trim().parse().unwrap_or_else(|_| {
                        println!("Invalid number!");
                        0.0
                    })
                } else {
                    0.0
                };

                let result = match choice {
                    1 => num1 + num2,
                    2 => num1 - num2,
                    3 => num1 * num2,
                    4 => {
                        if num2 == 0.0 {
                            println!("Cannot divide by zero!");
                            continue;
                        }
                        num1 / num2
                    }
                    5 => num1.powf(num2),
                    6 => num1.sqrt(),
                    7 => num1.ln(),
                    8 => num1.sin(),
                    9 => num1.cos(),
                    10 => num1.tan(),
                    _ => {
                        println!("Invalid choice!");
                        0.0
                    }
                };
                println!("Result: {}", result);
            }
            _ => println!("Invalid choice!"),
        }
    }
}

